lib/matplotlib/widgets.py

```diff
@@ -2006,6 +2005,38 @@ def set_visible(self, visible):
         for artist in self.artists:
             artist.set_visible(visible)
 
+    @property
+    def artists(self):
+        """Tuple of the artists of the selector."""
+        if getattr(self, '_handles_artists', None) is not None:
+            return (self._selection_artist, ) + self._handles_artists
+        else:
+            return (self._selection_artist, )
+
+    def set_props(self, **props):
+        """
+        Set the properties of the selector artist. See the `props` argument
+        in the selector docstring to know which properties are supported.
+        """
+        self._selection_artist.set(**props)
+        if self.useblit:
+            self.update()
+        self._props.update(props)
+
+    def set_handle_props(self, **handle_props):
+        """
+        Set the properties of the handles selector artist. See the
+        `handle_props` argument in the selector docstring to know which
+        properties are supported.
+        """
+        if not hasattr(self, '_handles_artists'):
+            raise NotImplementedError("This selector doesn't have handles.")
+        for handle in self._handles_artists:
+            handle.set(**handle_props)
+        if self.useblit:
+            self.update()
+        self._handle_props.update(handle_props)
```

Comment:
You need to normalize properties here (see normalize_kwargs), otherwise with e.g.
```python
from matplotlib.widgets import *
ax = gca()
ss = SpanSelector(ax, print, "horizontal", interactive=True)
ss.set_handle_props(lw=.5); ss.set_handle_props(linewidth=3); ss.direction = "vertical"
```
the handles' linewidth end up being 0.5, not 3 (also because `linewidth` is an explicit kwarg to Line2D whereas `lw` is implicitly handled as a generic property, but again I think the correct fix is to just normalize here.)

Likewise for props, probably.

Reply:
Yes, this should be fixed in 8f0e823. Thanks!